packages/core/src/dispatcher.ts

```diff
@@ -37,18 +38,21 @@ function messageTypeToString(type: MsgType): string {
  */
 export class Dispatcher {
   readonly messageBus: MessageBus
+  readonly recordCache: LRUMap<CID, any>
```

Comment:
I think this cache might need to use strings as the key rather than CID instances, otherwise I think it might fail to detect two CID instances of the same CID contents as equivalent (though there may be a way to provide a custom equality checker to LRUMap?  I'm not sure about that though).

Should be fairly straightforward to do a simple test to confirm though

Reply:
Since this also reduces the size of the Map, I just went ahead and changed the key to `string`.

Reply:
And you were absolutely right here - `LRUMap<CID, any>` with an entry for a CID failed to look that entry up with a different instance of the same CID.